src/lucky_record/needy_initializer.cr

```diff
@@ -53,6 +53,20 @@ module LuckyRecord::NeedyInitializer
       end
     end
 
+    def self.save!(
+        params,
+        {% if NEEDS.size > 0 %}
+          **needs
+        {% end %}
+      )
+      form = new(
+        params,
+        {% if NEEDS.size > 0 %}
+          **needs
+        {% end %}
+      ).save!
```

Comment:
At some point I think it would be good to extract a `run` macro https://crystal-lang.org/api/0.23.1/Crystal/Macros.html#run%28filename%2C%2Aargs%29%3AMacroId-instance-method so that we don't need to repeat the arguments list every time for each version of save. But I think this is fine for now. Here's another example of a run macro in case you're curious: https://github.com/luckyframework/web/blob/master/src/run_macros/infer_route.cr called from https://github.com/luckyframework/web/blob/master/src/lucky_web/routeable.cr#L47

Reply:
Yeah this started to feel not DRY. Also, I need to read more about macros.